components/plate_based_widgets/mapeditor/PlateMapWidget.vue

```diff
@@ -0,0 +1,737 @@
+      width="66"
+      height="66"
+    >
+      <PlateWell
+        :classname="'plate_' + well_index"
+        :svg_height="66"
+        :svg_width="66"
+        :circle_x="33"
+        :circle_y="33"
+        :radius="25"
+        :strk="hover_color[well_index]"
+        :plate_fill="platecolor[well_index]"
+        :stroke_wdth="stroke_width[well_index]"
+        :index="well_index"
+        @enter-well="on_wellenter(well_index)"
+        @leave-well="on_wellleave(well_index)"
+        @click-exact="basic_select(well_index)"
+        @click-shift-exact="basic_shift_or_ctrl_select(well_index)"
+      />
+    </span>
+  </div>
+</template>
+<script>
+import { library } from "@fortawesome/fontawesome-svg-core";
+import { faMinusCircle, faPlusCircle } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
+import PlateWell from "@/components/basic_widgets/PlateWell.vue";
+
+library.add(faMinusCircle);
+library.add(faPlusCircle);
+
+const no_stroke_width = 0;
+const hover_stroke_width = 2;
+const selected_stroke_width = 4;
+const hover_color = "#ececed";
+const selected_color = "#FFFFFF";
+const default_color = "#b7b7b7";
+const debug_mode = undefined;
+
+export default {
+  name: "PlateMap",
+  components: { FontAwesomeIcon, PlateWell },
+  props: {
+    selected: {
+      type: Array,
+      default: function () {
+        return new Array(24).fill(false);
+      },
+    },
+    platecolor: {
+      type: Array,
+      default: function () {
+        return new Array(24).fill(default_color);
+      },
+    },
+  },
+  data() {
+    return {
+      all_select_or_cancel: true,
+      x_origin: 0,
+      y_origin: 0,
+      rect_width: 0,
+      rect_height: 0,
+      all_select: this.selected,
+      hover: new Array(24).fill(false),
+      hover_color: new Array(24).fill(hover_color),
+      stroke_width: new Array(24).fill(no_stroke_width),
+      row_values: ["A", "B", "C", "D"],
+      column_values: ["1", "2", "3", "4", "5", "6"],
+      testerf: false,
+    };
+  },
+  watch: {
+    selected: function () {
+      // reset stroke width when selected wells is reset
+
+      if (this.selected.filter((x) => x).length === 0) {
+        this.all_select = Array(24).fill(false);
+        this.stroke_width = Array(24).fill(no_stroke_width);
+      }
+    },
+  },
+  created() {
+    this.stroke_width.splice(0, this.stroke_width.length);
+    for (let j = 0; j < this.all_select.length; j++) {
+      this.stroke_width[j] = !this.all_select[j] ? no_stroke_width : selected_stroke_width;
+    }
+    const allEqual = (arr) => arr.every((v) => v === true); // verify in the pre-select all via a const allEqual function.
+    this.all_select_or_cancel = allEqual(this.all_select) ? false : true; // if pre-select has all wells is true, then toggle from (+) to (-) icon.
+  },
+  methods: {
+    column_left_offset(column) {
+      switch (column) {
+        case "1":
```

Comment:
Oh they were actually just moved to this file. I think we should try to clean these up. I don't think anything after the decimal is really all that necessary so these can probably be rounded to start, and then there should be some formula we can apply instead of using a switch-case statement. Something like `36 + (column * 61)` should work

Reply:
Yeah this could definitely use cleaning, I basically refactored the stim studio platemap how it should work, so I could just move that all over here, there's like too much to clean up